spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/PubsubAdmin.java

```diff
@@ -46,7 +47,8 @@
 	private SubscriptionAdminClient subscriptionAdminClient;
 
 	/** Default inspired in the subscription creation web UI. */
-	private static final int DEFAULT_ACK_DEADLINE = 10;
+	@Value("${spring.cloud.gcp.pubsub.subscriber.ackDeadline:10}")
```

Comment:
??? So, in your terms anything with setter is evil?

All the components in Spring are based on setters. 
Right, you should always keep in mind if you can change the state of an object it isn't thread-safe.
That's not only for Spring classes.

I insist that we should not use `@Value` in target components. That is end-user application feature or maximum auto-config.

And why do you care too much that it won't be thread-safe?.. More over the story is only about an `int` value, which can be `volatile` and that's all...